src/gwt/src/org/rstudio/core/client/virtualscroller/VirtualScrollerManager.java

```diff
@@ -0,0 +1,106 @@
+package org.rstudio.core.client.virtualscroller;
+
+import com.google.gwt.core.client.JavaScriptException;
+
+import java.util.Date;
+import java.util.HashMap;
+import com.google.gwt.dom.client.Element;
+import org.rstudio.core.client.Debug;
+import org.rstudio.core.client.ExternalJavaScriptLoader;
+import org.rstudio.core.client.theme.res.ThemeStyles;
+
+public class VirtualScrollerManager
+{
+   public static void init()
+   {
+      if (!initialized_ && !loading_)
+      {
+         scrollers_ = new HashMap<>();
+         loading_ = true;
+         virtualScrollerLoader_.addCallback(() -> {
+            try
+            {
+               initialized_ = true;
+               loading_ = false;
+            }
+            catch (JavaScriptException e)
+            {
+               // If we get an error initializing the virtualScroller, log
+               // and no-op the error to continue non-virtualized
+               Debug.log(e.getMessage());
```

Comment:
Should we clear the `loading_` flag here? (doesn't look like it effects correctness but would be consistent?)